runtime/src/qio/qio_formatted.c

```diff
@@ -361,6 +361,86 @@ qioerr _append_char(char* restrict * restrict buf, size_t* restrict buf_len, siz
   return 0;
 }
 
+qioerr read_entire_file(const int threadsafe,qio_channel_t* restrict ch, const char* restrict* restrict out, int64_t* restrict len_out, ssize_t maxlen)
+{  
+  qioerr err,err_reading_file=0;
+  int64_t file_length;
+  char* ret=NULL;
+  ssize_t amt;
+  err_t errcode;
+  int64_t fileLenGuess=1024;
+  if( threadsafe ) 
+  {
+    err = qio_lock(&ch->lock);
+    if( err ) return err;
+  }
+
+  err = qio_channel_mark(false, ch);
+  if( err ) goto unlock;
+
+  //getting file length
+  err=qio_file_length(ch->file,&file_length);
+  //if file length successfully obtained
+  if(!err)
+  {
+    ret=qio_malloc(file_length+1);
+    ret[0]='\0';
+    err=qio_channel_read(false, ch, ret, file_length, &amt);
+  }
```

Comment:
If the file length grew while we did this read, the err will be 0 but we still need to loop.

Reply:
OK.Thanx

Reply:
@mppf  I checked and I found that qio_channel_read is not returning EEOF if the file read is completed . I am not sure if I am making any mistake. I checked in the original code for the qio_channel_read_string  function whether the value of err in line 463 becomes equal to EEOF after the read is completed but this condition (qio_err_to_int(err)==EEOF) is becoming false there.